src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -205,22 +206,35 @@ osexperimental_handle_download_packages (RPMOSTreeOSExperimental *interface,
   if (!rpmostree_context_download_metadata (ctx, DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_RPMDB,
                                             cancellable, &local_error))
     goto out;
-
   sack = dnf_context_get_sack (rpmostree_context_get_dnf (ctx));
-
-  for (const char* const* it = queries; it && *it; it++)
+  RpmOstreeOverrideSourceKind source_kind;
+  source_name = rpmostree_parse_override_source (source, &source_kind, &local_error);
+  if (local_error)
+    goto out;
+  if (source_kind == RPM_OSTREE_OVERRIDE_SOURCE_KIND_REPO)
     {
-      auto query = static_cast<const char*> (*it);
-      g_autoptr(GPtrArray) pkglist = rpmostree_get_matching_packages (sack, query);
-
-      if (!pkglist || pkglist->len == 0)
+      g_autoptr(GPtrArray) pkglist = NULL; 
```

Comment:
This should be inside the for-loop too. Otherwise we'll leak every iteration except the last one.